algorithm/algo_1.rs

```diff
@@ -0,0 +1,23 @@
+//Binary Search
+use std::convert::TryInto;
+impl Solution {
+    pub fn binary_search(list: Vec<i32>, item: i32) -> Option<i32> {
```

Comment:
Is this from LeetCode? If not, it's better to return `Option<usize>` since Rust always use `usize` for an index.

Reply:
I converted the Python code from the textbook to Rust. I will change it to return Option<usize>.